### ISSUE TITLE
1. Introduction to gradient methods and mathematical programming (2 weeks)

### ISSUE THREAD

**Issue by Lilas-B** [Lilas-B/Image_Analysis_Project#3]

**Comment by luka2g** [comment]
Please go through the tutorials which are listed
- https://www.toptal.com/python/gradient-descent-in-tensorflow
- https://www.tensorflow.org/guide/autodiff
- https://pytorch.org/tutorials/beginner/introyt/autogradyt_tutorial.html

Your tasks are 
- [ ] Read through the materials and make your notes
- [ ] Read the notes on how to use git (https://gitdailies.com/articles/link-github-commit-to-issue/; https://stackoverflow.com/questions/1687262/link-to-the-issue-number-on-github-within-a-commit-message; https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/)
- [ ] Submit (by a git commit) a jupyter notebook with your condensed notes and tests which you have used to study these two features of ML

We will then discuss the solution in person and decide on closing the issue
